<!-- -->
<launch>
  <!-- Ego vehicle -->
  <arg name='host' default='localhost'/>
  <arg name='port' default='2000'/>
  <arg name='role_name' default='ego_vehicle'/>
  <arg name='config_file' default=''/>
  <arg name="vehicle_filter" default='vehicle.*'/>
  <arg name="spawn_point" default=""/><!-- use comma separated format "x,y,z,roll,pitch,yaw" -->
  <arg name="track" default="t1_triple" />
  <arg name="model_free" default="1" />
  <arg name="offscreen" default="false" />
  <arg name="scenario" default="true" />
  <arg name="vis2D" default="false" />

  <!-- spwan the ego vehicle -->
  <include file="$(find carla_spawn_objects)/launch/carla_spawn_objects.launch">
    <arg name="objects_definition_file" value='$(arg config_file)'/>
    <arg name="spawn_point_ego_vehicle" value="$(arg spawn_point)"/>
    <arg name="spawn_sensors_only" value="false" />
  </include>

  <!-- carla_ackermann_control nodes -->
  <include file="$(find carla_ackermann_control)/launch/carla_ackermann_control.launch">
    <arg name='role_name' value='$(arg role_name)' />
  </include>

  <!-- carla manual control -->
  <group unless="$(arg offscreen)">
      <node pkg="graic_core" type="carla_manual_control.py" name="carla_manual_control_$(arg role_name)" output="screen">
        <param name="role_name" value="$(arg role_name)"/>
      </node>
      <!-- record video -->
      <node pkg="graic_core" type="video_output.py" name="video_output_$(arg role_name)" output="screen">
        <param name="role_name" value="$(arg role_name)"/>
      </node>
  </group>

  <!-- the perception module -->
  <include file="$(find graic_perception)/launch/graic_perception.launch">
    <arg name='role_name' value='$(arg role_name)' />
    <arg name='host' value='$(arg host)' />
    <arg name='port' value='$(arg port)' />
  </include>

  <!-- the scorer module -->
  <include file="$(find graic_scorer)/launch/graic_scorer.launch">
    <arg name='host' value='$(arg host)' />
    <arg name='port' value='$(arg port)' />
    <arg name='role_name' value='$(arg role_name)' />
    <arg name='track' value='$(arg track)' />
    <arg name='model_free' value='$(arg model_free)' />
  </include>

  <!-- the waypoints module -->
  <include file="$(find graic_waypoints)/launch/graic_waypoints.launch">
    <arg name='host' value='$(arg host)' />
    <arg name='port' value='$(arg port)' />
    <arg name='role_name' value='$(arg role_name)' />
    <arg name='track' value='$(arg track)' />
  </include>

  <!-- the scenario module -->

</launch>
